code-challenges.js

```diff
@@ -45,16 +94,64 @@ var stringWithVowels2 = "ILoveJavaScript"
+const noVowels1 = (object) => {
+    //declare variable that contains all vowels
+    let vowels = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]
+    //if statement to run the function that will return a string with no vowels if a string is entered
+    if (typeof object === "string") {
+    //declare variable that will split the string into an array of substrings
+        let arr = object.split("")
+    //use filter method to go through variable arr of substrings and take out the vowels present in input (indicated by -1) and then join cohesively at the end using join method
+            return typeof arr.filter(value => vowels.indexOf(value) === -1).join("")
+    //else statement to declare not a string if a boolean or number is added
+     } else {
+        return "not a string"
+     }
+    }
+    //print "not a string" if variable doesn't contain a string
+    console.log(noVowels1(notAString1))
+    console.log(noVowels1(notAString2))
+
 
 
 
 // --------------------6) Create a function that takes the toonimals array and returns an array with only the toon objects that are cats.
 // Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
 
-var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
-
-
+var toonimals = [ 
+    {name: "Itchy", animal: "mouse"}, 
+    {name: "Stimpy", animal: "cat"}, 
+    {name: "Daffy", animal: "duck"}, 
+    {name: "Scratchy", animal: "cat"}, 
+    {name: "Ren", animal: "dog"}, 
+    {name: "Felix", animal: "cat"}
+]
+//declare function that takes in an object
+const onlyCats = (object) => {
+    //use filter method to go through all values in object and only return the values who contain "cat"
+     return object.filter(value => value.animal === "cat")
+}
+//print the name and aninaml type for cats only
+console.log(onlyCats(toonimals))
 
 // --------------------7) Using the toonimals variable from #6, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+var toonimals1 = [ 
+    {name: "Itchy", animal: "mouse"}, 
+    {name: "Stimpy", animal: "cat"}, 
+    {name: "Daffy", animal: "duck"}, 
+    {name: "Scratchy", animal: "cat"}, 
+    {name: "Ren", animal: "dog"}, 
+    {name: "Felix", animal: "cat"}
+]
+//declare a function that is taking in an array
+const noCats = (array) => {
+//create a filter to get rid of all animals that have cat under the animal value in object
+    let getRid = array.filter(value => value.animal !=="cat")
+//use .map function to go through what values are left after getRid return the names of the cat less animals and join method them so they display the names without being in an array
+    return getRid.map(value => value.name).join(" ")
+}
+console.log(noCats(toonimals1))
```

Comment:
good job! you can even chain the .map to the end of the .filter if you'd like! yay!